src/amo/components/LandingPage.js

```diff
@@ -41,9 +44,33 @@ export class LandingPageBase extends React.Component {
 
   static defaultProps = {
     apiAddonType: getApiAddonType,
+    contentForType: null,
   }
 
-  contentForType(visibleAddonType) {
+  componentWillMount() {
+    this.setViewContextType();
+  }
+
+  componentDidUpdate() {
+    this.setViewContextType();
+  }
+
+  setViewContextType() {
+    const { apiAddonType, dispatch, params } = this.props;
+
+    try {
+      const addonType = apiAddonType(params.visibleAddonType);
+      dispatch(setViewContext(addonType));
+    } catch (err) {
+      if (err instanceof AddonTypeNotFound) {
+        log.info('AddonTypeNotFound in setViewContextType()', err);
```

Comment:
The site still works if this error gets thrown, right? I think this should throw an error, especially because I don't want us to forget about it when we fix the routing issue.